### PR TITLE
Order filter - Expiring subscription search

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
Option to see the Orders which are expiring on Future Days/Dates

### CODE REVIEW

File: app/Http/Controllers/Order/OrderSearchController.php [hunk 1]
```diff
@@ -39,9 +42,12 @@ public function advanceOrderSearch(Request $request)
             $this->domain($request->input('domain'), $baseQuery);
             $this->allInstallations($request->input('act_ins'), $baseQuery);
             $this->allRenewals($request->input('renewal'), $baseQuery);
-            $this->getSelectedVersionOrders($baseQuery, $request->input('version'), $request->input('product_id'));
-
-            return $baseQuery->orderBy('orders.created_at', 'desc');
+            $this->getSelectedVersionOrders($baseQuery, $request->input('version'), $request->input('product_id'), $request);
+            if ($request->renewal == 'expiring_subscription') {
```
Comment (RafficMohammed) [inline app/Http/Controllers/Order/OrderSearchController.php:46]: use ternary operator

File: app/Http/Controllers/Order/OrderSearchController.php [hunk 2]
```diff
@@ -238,6 +244,11 @@ private function product($product_id, $join)
      */
     public function orderFrom($till, $from, $join)
     {
+        if ($request->renewal = 'expiring_subscription') {
```
Comment (RafficMohammed) [inline app/Http/Controllers/Order/OrderSearchController.php:247]: use ternary

File: app/Http/Controllers/Order/OrderSearchController.php [hunk 3]
```diff
@@ -262,14 +273,19 @@ public function orderFrom($till, $from, $join)
      */
     public function orderTill($from, $till, $join)
     {
+        if ($request->renewal = 'expiring_subscription') {
+            $subTo = 'subscriptions.update_ends_at';
```
Comment (RafficMohammed) [inline app/Http/Controllers/Order/OrderSearchController.php:277]: here too

**Review by RafficMohammed — APPROVED** [review]

**Comment by bhanu2217** [comment]
Is this P.R for this issue? https://github.com/ladybirdweb/agora-invoicing-community/issues/1930

**Review by sandesh556 — APPROVED** [review]

**Comment by sandesh556** [comment]
@sowmiyasubramaniyan rebase this

**Review by sandesh556 — APPROVED** [review]

File: app/Http/Controllers/Order/OrderSearchController.php [hunk 4]
```diff
@@ -31,17 +31,22 @@ public function __construct()
...
+            $this->orderTill($request->input('from'), $request->input('till'), $baseQuery, $request->renewal);
             $this->domain($request->input('domain'), $baseQuery);
             $this->allInstallations($request->input('act_ins'), $baseQuery);
             $this->allRenewals($request->input('renewal'), $baseQuery);
-            $this->getSelectedVersionOrders($baseQuery, $request->input('version'), $request->input('product_id'));
 
-            return $baseQuery;
+            $this->getSelectedVersionOrders($baseQuery, $request->input('version'), $request->input('product_id'), $request);
+
+            return $request->renewal == 'expiring_subscription' ? $baseQuery->orderBy('subscriptions.update_ends_at', 'asc') :
```
Comment (RafficMohammed) [inline app/Http/Controllers/Order/OrderSearchController.php:48]: check this

**Review by RafficMohammed — CHANGES_REQUESTED** [review]
Check ternary operator

**Review by sandesh556 — DISMISSED** [review]

**Review by priyadarshini2101 — DISMISSED** [review]
Option to see the Orders which are expiring on Future Days/Dates
After fix

feb7af1

**Review by SarfrazS — DISMISSED** [review]
Working fine.

**Comment by sandesh556** [comment]
Rebase this PR properly

**Pull request closed without merging**